### DISCUSSION TITLE
`findUnique` always returning `null` on a `Json` field

### DISCUSSION THREAD

**Discussion by baptistemarchand** [prisma/prisma#20367] [Q&A] [answered]
Hi,
It seems like doing a findUnique on a Json field is not working. Is that a known limitation?

My schema looks something like this :
```prisma
model Address {
  id        String
  geopoint  Json  @unique(map: "geopoint_UNIQUE")
}
```
And when I do :
```js
const geo = {lat: 48.872147,lng: 2.42892}

console.log('findUnique', await prisma.address.findUnique({where: {geopoint: {equals: geo}}}))

console.log('findFirst', await prisma.address.findFirst({where: {geopoint: {equals: geo}}}))
```

I get the following result :
```
findUnique null
findFirst { id: 'ckz3yuim000om0990pmrwm5in', ...}
```

It's not really blocking me as I can just do a `findFirst` but I was lucky to notice this before it got to production. It would be hard to debug because this behaviour is quite surprising.

**Comment by ludralph** [comment]
Hi @baptistemarchand 👋 

Thank you for raising this question. I can also reproduce this behavior, but I'm unsure if it is the expected behavior. 

Can you please open an issue using this [link](https://github.com/prisma/prisma/issues/new?assignees=&labels=kind%2Fbug&template=bug_report.yml) so that our engineers can take a look?

**Comment by Jolg42** [comment, marked as answer]
@baptistemarchand 

You can change your `findUnique` query 
from:
```ts
await prisma.address.findUnique({where: {geopoint: {equals: geo}})
```
to:
```ts
await prisma.address.findUnique({where: {geopoint: geo}})
```

What happens is that filters, here `equals` are not supported by `findUnique` and the query sent to Postgres is
```SQL
SELECT "public"."Address"."geopoint", "public"."Address"."id" FROM "public"."Address" WHERE ("public"."Address"."geopoint"::jsonb = $1 AND 1=1) LIMIT $2 OFFSET $3
$1 = '{"equals": {"lat": 48.872147, "lng": 2.42892}}', $2 = '1', $3 = '0'
```
See how it's asking Postgres to find `'{"equals": {"lat": 48.872147, "lng": 2.42892}}'` in your database which causes this problem here.

We have proper type/runtime validation errors when the field in not Json at the moment.
In the future, we might be able to improve this but with the current API it is not easy as technically you might want to match some JSON value or use a filter, and we can't tell for sure. One thought we had previously is https://github.com/prisma/prisma/issues/19611, but it's a serious breaking change.
We can start by better docs for now.

**Reply by baptistemarchand** [reply]
Thank you for the details Joel.

**Reply by Jolg42** [reply]
I created an issue for docs: https://github.com/prisma/docs/issues/5098
